src/Features/Core/Portable/Completion/ArgumentContext.cs

```diff
@@ -29,16 +30,42 @@ internal sealed class ArgumentContext
 
         internal ArgumentProvider Provider { get; }
 
-        public Document Document { get; }
+        /// <summary>
+        /// Gets the semantic model where argument completion is requested.
+        /// </summary>
+        public SemanticModel SemanticModel { get; }
```

Comment:
Why the switch to SemanticModel?

Reply:
AnalyzerRunner was going to need it anyway. `Document` wasn't providing much.